fixtures/dummy/all.go

```diff
@@ -130,114 +169,872 @@ func PopulateDBWithDummyModels(gormDB *gorm.DB) error {
+		ConfigAnalyses:               append([]models.ConfigAnalysis{}, AllDummyConfigAnalysis...),
+		ConfigComponentRelationships: append([]models.ConfigComponentRelationship{}, AllDummyConfigComponentRelationships...),
+		Teams:                        append([]models.Team{}, AllDummyTeams...),
+		Incidents:                    append([]models.Incident{}, AllDummyIncidents...),
+		Hypotheses:                   append([]models.Hypothesis{}, AllDummyHypotheses...),
+		Evidences:                    append([]models.Evidence{}, AllDummyEvidences...),
+		Canaries:                     append([]models.Canary{}, AllDummyCanaries...),
+		Checks:                       append([]models.Check{}, AllDummyChecks...),
+		CheckStatuses:                append([]models.CheckStatus{}, AllDummyCheckStatuses...),
+		Responders:                   append([]models.Responder{}, AllDummyResponders...),
+		Comments:                     append([]models.Comment{}, AllDummyComments...),
+		CheckComponentRelationships:  append([]models.CheckComponentRelationship{}, AllDummyCheckComponentRelationships...),
+	}
+
+	return d
+}
+
+// GenerateDynamicDummyData is similar to GetStaticDummyData()
+// except that the ids are randomly generated on call.
+func GenerateDynamicDummyData() DummyData {
+	var (
+		DummyCreatedAt   = time.Date(2022, time.December, 31, 23, 59, 0, 0, time.UTC)
+		DummyYearOldDate = time.Now().AddDate(-1, 0, 0)
+	)
+
+	// People
+	var JohnDoe = models.Person{
+		ID:   uuid.New(),
+		Name: "John Doe",
+	}
+
+	var JohnWick = models.Person{
+		ID:   uuid.New(),
+		Name: "John Wick",
+	}
+
+	var people = []models.Person{JohnDoe, JohnWick}
+
+	// Agents
+	var GCPAgent = models.Agent{
+		ID:   uuid.New(),
+		Name: "GCP",
+	}
+
+	var agents = []models.Agent{
+		GCPAgent,
+	}
+
+	// Teams
+	var BackendTeam = models.Team{
+		ID:        uuid.New(),
+		Name:      "Backend",
+		Icon:      "backend",
+		CreatedBy: JohnDoe.ID,
+		CreatedAt: time.Now(),
+		UpdatedAt: time.Now(),
+	}
+
+	var FrontendTeam = models.Team{
+		ID:        uuid.New(),
+		Name:      "Frontend",
+		Icon:      "frontend",
+		CreatedBy: JohnDoe.ID,
+		CreatedAt: time.Now(),
+		UpdatedAt: time.Now(),
+	}
+
+	var teams = []models.Team{BackendTeam, FrontendTeam}
+
+	// Topologies
+	var LogisticsTopology = models.Topology{
+		ID:        uuid.New(),
+		Name:      "logistics",
+		Namespace: "default",
+	}
+
+	var topologies = []models.Topology{
+		LogisticsTopology,
+	}
+
+	// Components
+	var Logistics = models.Component{
+		ID:         uuid.New(),
+		Name:       "logistics",
+		Type:       "Entity",
+		ExternalId: "dummy/logistics",
+		Labels:     types.JSONStringMap{"telemetry": "enabled"},
+		Owner:      "logistics-team",
+		CreatedAt:  DummyCreatedAt,
+		Status:     types.ComponentStatusHealthy,
+	}
+
+	var LogisticsAPI = models.Component{
+		ID:         uuid.New(),
+		Name:       "logistics-api",
+		ExternalId: "dummy/logistics-api",
```

Comment:
Wouldn't this lead to us having to update fixtures in 2 places ?

Reply:
For all the resources

Reply:
We don't have to update the fixture in 2 places. The old dummy data and ones in this func are completely isolated and can grow up/down independently.

Reply:
Then why use duplicated values ? This just feels confusing & needlessly complicated